### PR TITLE
Started my keymap and rgb code

### PR DESCRIPTION
**Pull request by AreYouConfused**
## Description

I have added my custom keymap and macro template

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/melody96/keymaps/areyouconfused/melody96.h [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (zvecr) [inline keyboards/melody96/keymaps/areyouconfused/melody96.h:1]: Really not a fan of this addition, as its got a lot of potential to cause issues (and not a paradigm that would be clean if it propagates through the repo). It relies on replacing an existing implementation based on QMKs lax use of VPATH, which would need to be kept in step with any renames and such.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/melody96/keymaps/areyouconfused/config.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#pragma once
```
Comment (drashna) [inline keyboards/melody96/keymaps/areyouconfused/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/melody96/keymaps/areyouconfused/custom.c [hunk 3]
```diff
@@ -0,0 +1,128 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/melody96/keymaps/areyouconfused/custom.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/melody96/keymaps/areyouconfused/melody96.c [hunk 4]
```diff
@@ -0,0 +1 @@
+#include "melody96.h"
```
Comment (drashna) [inline keyboards/melody96/keymaps/areyouconfused/melody96.c:1]: Please remove this file

File: keyboards/melody96/keymaps/areyouconfused/melody96.h [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#include "../../melody96.h"
```
Comment (drashna) [inline keyboards/melody96/keymaps/areyouconfused/melody96.h:1]: Please remove this file

File: keyboards/melody96/keymaps/areyouconfused/my_keycodes.h [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#pragma once
```
Comment (drashna) [inline keyboards/melody96/keymaps/areyouconfused/my_keycodes.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/melody96/keymaps/areyouconfused/keymap.c [hunk 7]
```diff
@@ -0,0 +1,72 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/melody96/keymaps/areyouconfused/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**